src/httpserver.h

```diff
@@ -124,21 +126,10 @@ class HTTPRequest
      * main thread, do not call any other HTTPRequest methods after calling this.
      */
     void WriteReply(int nStatus, const std::string& strReply = "");
-};
 
-/** Get the query parameter value from request uri for a specified key, or std::nullopt if the key
- * is not found.
- *
- * If the query string contains duplicate keys, the first value is returned. Many web frameworks
- * would instead parse this as an array of values, but this is not (yet) implemented as it is
- * currently not needed in any of the endpoints.
- *
- * Helper function for HTTPRequest::GetQueryParameter.
- *
- * @param[in] uri is the entire request uri
- * @param[in] key represents the query parameter of which the value is returned
- */
-std::optional<std::string> GetQueryParameterFromUri(const char* uri, const std::string& key);
+    /** Get libevent parsed URI, which is nullptr if uri parsing failed */
+    evhttp_uri* GetURIParsed();
```

Comment:
```suggestion
    const evhttp_uri* GetURIParsed() const;
```

Reply:
Yeah, putting it back, before the compiler was complaining as in the previous approach we were modifying m_uri_parsed in this function.